### ISSUE TITLE
SVG opacity doesn't seem to work

### ISSUE THREAD

**Issue by JonathanReeve** [qutebrowser/qutebrowser#3061]
I use a web app in the course management system Canvas called SpeedGrader. Speedgrader allows you to annotate PDFs using highlight annotations. This works fine in Firefox and Chromium, but in qutebrowser, text highlights are opaque, so I can't see the word that's been highlighted (it's yellow censorship, really). 

I used the web inspector to see what kind of HTML is responsible, and it's `<svg><g><rect fill-opacity="1">`. 

I don't understand why qutebrowser would handle this kind of thing differently from Chromium, but it does. 

Version info

qutebrowser v0.11.0
Git commit: 
Backend: QtWebEngine (Chromium 56.0.2924.122)

CPython: 3.6.2
Qt: 5.9.1
PyQt: 5.9

sip: 4.19.3
colorama: no
pypeg2: 2.15
jinja2: 2.9.6
pygments: 2.2.0
yaml: 3.12
cssutils: no
typing: yes
OpenGL: 3.1.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebKitWidgets: yes
pdf.js: 1.5.188 (/home/jon/.local/share/qutebrowser/pdfjs/build/pdf.js)
SSL: OpenSSL 1.0.2l  25 May 2017

Style: QFusionStyle
Platform: Linux-4.12.13-1-ARCH-x86_64-with-arch-17.6-ISO-Rolling, 64bit
Linux distribution: Antergos Linux (unknown)
Frozen: False
Imported from /usr/lib/python3.6/site-packages/qutebrowser
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt
OS Version: 

--- /etc/os-release ---
NAME="Antergos Linux"
VERSION="17.6-ISO-Rolling"
ID="antergos"
ID_LIKE="arch"
PRETTY_NAME="Antergos Linux"
CPE_NAME="cpe:/o:antergosproject:antergos:17.6"

--- /etc/arch-release ---
Antergos Linux release 17.6 (ISO-Rolling)

Paths:
config: /home/jon/.config/qutebrowser
data: /home/jon/.local/share/qutebrowser
system_data: /usr/share/qutebrowser
cache: /home/jon/.cache/qutebrowser
download: /home/jon/Downloads
runtime: /run/user/1000/qutebrowser

**Comment by The-Compiler** [comment]
`opacity="1"` means fully opaque, so I don't see how that'd be transparent. Do you have an example page where I can easily reproduce it? FWIW, on https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/opacity I see a black and a grey circle, like expected.

**Comment by JonathanReeve** [comment]
Yeah, I don't get it. I try to reproduce this in codepen, and I can't, really. But this is what it looks like in the app: 

![screenshot from 2017-10-07 09-07-38](https://user-images.githubusercontent.com/1843676/31308270-e61af5ce-ab41-11e7-8d0e-db0f859bc807.png)

And here's what it looks like in Firefox: 

![screenshot from 2017-10-07 09-08-41](https://user-images.githubusercontent.com/1843676/31308271-eb7e2482-ab41-11e7-961d-44571af7164d.png)

The HTML for a highlight annotation like that looks like this: 

```
<svg aria-selected="true" aria-labelledby="title" class="Annotation-svg HighlightAnnotation isAnnotationFocused" style="width: 36.41px; height: 29.51px;"><title>Highlight annotation</title><g><rect class="SelectionBox" x="2" y="2" width="32.40999999999997" height="25.510000000000105" fill-opacity="0" stroke="#bbb" stroke-dasharray="3, 3" stroke-width="1" stroke-opacity="1"></rect></g><g role="presentation" fill="#FCE680" fill-opacity="1"><rect x="4" y="5.7549999999999955" width="9.49499999999989" height="18.000000000000114"></rect><rect x="11.904999999999973" y="5.7549999999999955" width="1.5899999999999181" height="18.000000000000114"></rect><rect x="4" y="5.7549999999999955" width="9.49499999999989" height="18.000000000000114"></rect><rect x="11.904999999999973" y="5.7549999999999955" width="1.5899999999999181" height="18.000000000000114"></rect><rect x="11.904999999999973" y="5.7549999999999955" width="20.504999999999995" height="18.000000000000114"></rect><rect x="11.904999999999973" y="4" width="20.504999999999995" height="21.510000000000105"></rect></g></svg>
```

I thought it probably has something to do with `fill-opacity`, but maybe it has something to do with z-index, or maybe some Javascript is setting opacity after the fact? 

I'll keep trying to reproduce this on a public site, since Canvas, as a course management tool, isn't public.

**Comment by JonathanReeve** [comment]
Also, FWIW, the problem doesn't seem to appear using Webkit, only Webengine.

**Comment by The-Compiler** [comment]
Can you try with [QupZilla](https://qupzilla.com/) please?

**Comment by JonathanReeve** [comment]
Same problem in QupZilla, it looks like.

**Comment by The-Compiler** [comment]
Looks like it's a QtWebEngine issue, and not a qutebrowser issue then. It might be fixed in Qt 5.10 already (which should be released early December), as that updates the underlying Chromium from 56 to 60 (or maybe 61). Unfortunately I don't know of an easy way to try that yourself, other than compiling things from sources.

I suggest you wait until Qt 5.10 landed in Arch (and thus Antergos if I'm not mistaken), and if it's not fixed until then, open a bug [in the Qt bugtracker](https://bugreports.qt.io/).
